Model/Sync/Catalog/Processor.php

```diff
@@ -698,21 +698,31 @@ protected function existDataRequest($url, $requestIds, $sqlData, $type = 'produc
 
     /**
      * Prepare products to sync their category data
-     * @param array<mixed> $data
-     * @param array<mixed> $collectionItems
+     * @param array<mixed> $productsSyncAttempt
+     * @param array<mixed> $collectionItemsToSync
      * @param array<mixed> $dataForCategorySync
      * @return array<mixed>
      */
-    protected function getProductsForCategorySync($data, $collectionItems, array $dataForCategorySync): array
+    protected function getProductsForCategorySync($productsSyncAttempt, $collectionItemsToSync, array $dataForCategorySync): array
     {
-        foreach ($data as $dataItem) {
-            foreach ($collectionItems as $item) {
-                if ($item->getId() == $dataItem['product_id']) {
-                    $dataForCategorySync[$dataItem['yotpo_id']] = $item;
-                    break;
-                }
+        $storeId = $this->coreConfig->getStoreId();
+        $retryItems = [];
+        if ($this->retryItems && isset($this->retryItems[$storeId])) {
```

Comment:
This variable is initialized, why do you need to check if it exists `$this->retryItems`?
```
/**
     * @var array<mixed>
     */
    protected $retryItems = [];
```